cmd/create/machinepool/cmd.go

```diff
@@ -207,6 +210,18 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// Validate flags that are only allowed for multi-AZ clusters
+	isMultiAvailabilityZoneSet := cmd.Flags().Changed("multi-availability-zone")
+	if isMultiAvailabilityZoneSet && !cluster.MultiAZ() {
+		reporter.Errorf("Setting the `multi-availability-zone` flag is only allowed for multi-AZ clusters")
+		os.Exit(1)
+	}
+	isAvailabilityZoneSet := cmd.Flags().Changed("availability-zone")
+	if isAvailabilityZoneSet && !cluster.MultiAZ() {
+		reporter.Errorf("Setting the `availability-zone` flag is only allowed for multi-AZ clusters")
+		os.Exit(1)
+	}
+
```

Comment:
also check if `isMultiAvailabilityZoneSet` and isMultiAvailabilityZone==true and isAvailabilityZoneSet --> not allowed

Reply:
Added.
![image](https://user-images.githubusercontent.com/57869309/172600065-b38bf2d6-a706-40f0-a36c-29d9bf77eebe.png)